Mlem/Views/Tabs/Posts/Components/Comment/Components/Comment Item.swift

```diff
@@ -195,19 +205,33 @@ struct CommentItem: View
     }
     
     private func rate(_ comment: HierarchicalComment, operation: ScoringOperation) async throws {
+        guard localVote == nil else {
+            // if we have a local vote then we're in the middle of rating
+            // so avoid the user being able to initiate additional requests
```

Comment:
This could result in a bug where an additional vote (such as a downvote after accidentally pressing up, or even resetting the vote) would be discarded, is there a way to queue this up, or something?

Reply:
Possibly a "deferredAction" that'll get replaced atomically when this function is called, and taken atomically at the end of the rating function, to prevent data races

Reply:
This is true, however I think any approach like that would be best tackled as part of #117, so we can have a unified system that works across posts and comments... arguably it could even be an abstraction that allows for any type of deferred set of actions?

There is some thought needed as well as to what our state should be, as we'd need to predict the new state on each press, but defer the call till we know the outcome of the in-flight one, and on a patchy network or overloaded server, what if three calls queue-up and one fails? What if some blip caused them to return in a different order? How many should we queue, as the user could just keep tapping indefinitely and spam the API 😬 

My preference would be that we disable the interaction while a call is in-flight, there is a `.disabled(...)` modifier in `SwiftUI` which I initially added, but that flips the colour of the buttons to being greyed out which looked a bit... iffy, since the call is normally returned quickly it flicks between standard/greyed out so quickly it looked glitchy 😞 

In terms of the current UX with the early return/guard, there is a small haptic feedback triggered when the user taps on the button to indicate they've submitted a rating, this happens when the call initiates so in the above case no haptic feedback occurs which I think (for now) indicates to the user that the second/third/fourth presses are not registering.

Perhaps design could consider how best to communicate a call is in-flight, perhaps a small visual indicator is what we need to resolve the above without adding more complexity?

Reply:
I think we could ignore this then, and add it to the known bugs for the next release.